index.js

```diff
@@ -81,7 +81,7 @@ function _parseInput (input, opts, cb) {
 
   var commonPrefix = null
   input.forEach(function (item, i) {
-    if (typeof item === 'string') return
+    if (typeof item === 'string' || (Buffer.isBuffer(item) && !item.name)) return
```

Comment:
Okay, for simplicity the second approach is used and now automatically assign names to files without one. Guess since people often use variables for their buffers/streams defining an additional identifier shouldn't be required by them (otherwise its still possible via `opts.name` or manually).